src/components/model/ModelAssociations.svelte

```diff
@@ -0,0 +1,310 @@
+        return specA.priority - specB.priority;
+      }
+
+      return specA.id - specB.id;
+    });
+
+  function onClose() {
+    dispatch('close');
+  }
+
+  function onUpdatePriority(event: CustomEvent<{ id: number; priority: number }>) {
+    const {
+      detail: { id, priority },
+    } = event;
+    dispatch('updateSpecifications', {
+      ...selectedSpecifications[id],
+      id,
+      priority,
+    });
+  }
+
+  function onUpdateRevision(event: CustomEvent<{ id: number; revision: number | null }>) {
+    const {
+      detail: { id, revision },
+    } = event;
+    dispatch('updateSpecifications', {
+      ...selectedSpecifications[id],
+      id,
+      revision,
+    });
+  }
+
+  function onSave() {
+    dispatch('save');
+  }
+
+  function onSelectAssociation(event: CustomEvent<{ id: RadioButtonId }>) {
+    const {
+      detail: { id },
+    } = event;
+    selectedAssociationId = id as Association;
+    dispatch('selectAssociation', id as Association);
+  }
+
+  function onSelectSpecification(
+    event: CustomEvent<{
+      id: number;
+      revision: number | null;
+    } | null>,
+  ) {
+    selectedSpecification = event.detail;
+  }
+
+  function onSelectView(event: CustomEvent<{ id: RadioButtonId }>) {
+    const {
+      detail: { id },
+    } = event;
+    selectedViewId = id;
+  }
+</script>
+
+<div class="associations-container">
+  <div class="associations-header">
+    <div class="associations-title">Associations</div>
+    <RadioButtons selectedButtonId={selectedAssociationId} on:select-radio-button={onSelectAssociation}>
+      <RadioButton id="constraint"><div class="association-button">Constraints</div></RadioButton>
+      <RadioButton id="goal"><div class="association-button">Goals</div></RadioButton>
+      <RadioButton id="condition"><div class="association-button">Conditions</div></RadioButton>
+    </RadioButtons>
+    <div class="action-buttons">
+      <button class="st-button secondary w-100" on:click={onClose}> Close </button>
+      <button
+        class="st-button w-100"
+        disabled={!hasModelChanged}
+        on:click={onSave}
+        use:permissionHandler={{
+          hasPermission: hasEditSpecPermission,
+          permissionError: 'You do not have permission to update this model.',
+        }}
+      >
+        {hasModelChanged ? 'Save' : 'Saved'}
+      </button>
+    </div>
+  </div>
+  <CssGrid class="associations-css-grid" columns="1fr 3px 1fr">
+    <div class="associations-content">
+      <div class="associations-view">
+        <RadioButtons selectedButtonId={selectedViewId} on:select-radio-button={onSelectView}>
+          <RadioButton id="model"><div class="association-button">Model</div></RadioButton>
+          <RadioButton id="library"><div class="association-button">Library</div></RadioButton>
+        </RadioButtons>
+      </div>
+      {#if selectedViewId === 'library'}
+        <ModelSpecification
```

Comment:
Not seeing the empty state message in the Model subsection when i go to Library, have one constraint selected, deselect that constraint, and go to Model. 

https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/34c7c06b-f87b-4a4d-8afd-1f620256766c